### PR TITLE
Add export of all available reward points

### LINKED ISSUE #2020: Export Reward Point Summary
A new button `Export reward points` should be added to the right of the text showing the number of available reward points on the staff reward points redemption events page. Clicking the button should download a CSV file containing a summary of the reward points.

This file should contain the two columns `Email` and `Points` for each user, listing the number of points currently available for that user (grantings minus redemptions) next to the user's email address. A line should only be added for users where this number of available points is not zero.

### PR DESCRIPTION
**Pull request by PFischbeck**
This resolves #2020, as it adds a button to export a simple CSV that is an overview over all available reward points grouped by user profile.

Feel free to suggest better path names, or a way to improve the placement and look of the button:
grafik

### CODE REVIEW

File: evap/rewards/views.py [hunk 1]
```diff
@@ -158,6 +159,37 @@ def reward_point_redemption_event_export(request, event_id):
...
+@manager_required
+def reward_points_export(request):
+    filename = _("RewardPoints") + f"-{get_language()}.csv"
+    response = AttachmentResponse(filename, content_type="text/csv")
+
+    writer = csv.writer(response, delimiter=";", lineterminator="\n")
+    writer.writerow(
+        [
+            _("Email address"),
+            _("Number of points"),
```
Comment (niklasmohrin) [inline evap/rewards/views.py:171]: Removing the trailing comma in the list here (and also in the `for profile` loop below) will make our formatter put this in one line, which I would prefer :)

**Review by niklasmohrin — APPROVED** [review]
Works as expected, some suggestions

File: evap/rewards/views.py [hunk 2]
```diff
@@ -158,6 +159,37 @@ def reward_point_redemption_event_export(request, event_id):
...
+        ]
+    )
+    profiles_with_points = (
+        UserProfile.objects.annotate(
+            points=Sum("reward_point_grantings__value", default=0) - Sum("reward_point_redemptions__value", default=0)
+        )
+        .filter(points__gt=0)
+        .order_by("-points")
```
[the remark this answers is not in the document]
Comment (janno42) [reply evap/rewards/views.py:179]: I would say its useful to have the entries ordered, but wouldn't require a test for it.

File: evap/rewards/templates/rewards_reward_point_redemption_events.html [hunk 3]
```diff
@@ -13,6 +13,7 @@
         </div>
         <div class="col text-end">
             {% translate 'Available reward points' %}: {{ total_points_available }}
+            <a href="{% url 'rewards:reward_points_export' %}" class="btn btn-dark btn-sm ms-2">{% translate 'Export reward points' %}</a>
```
Comment (janno42) [inline evap/rewards/templates/rewards_reward_point_redemption_events.html:16]:
```suggestion
            <a href="{% url 'rewards:reward_points_export' %}" class="btn btn-light btn-sm ms-2">{% translate 'Export reward points' %}</a>
```

Please use a `btn-light`. `btn-dark` is currently used for actions that create/add something.

**Review by janno42 — CHANGES_REQUESTED** [review]

File: evap/rewards/tests/test_views.py [hunk 4]
```diff
@@ -189,6 +189,30 @@ def setUpTestData(cls):
         cls.url = f"/rewards/reward_point_redemption_event/{event.pk}/export"
 
 
+class TestPointsExportView(WebTestStaffModeWith200Check):
+    @classmethod
+    def setUpTestData(cls):
+        cls.test_users = [make_manager()]
+        cls.url = "/rewards/reward_points_export"
```
[the remark this answers is not in the document]
Comment (PFischbeck) [reply evap/rewards/tests/test_views.py]: Done!

File: evap/rewards/tests/test_views.py [hunk 5]
```diff
@@ -189,6 +189,30 @@ def setUpTestData(cls):
...
+
+        cls.student = baker.make(UserProfile, email="student@institution.example.com")
+        cls.event = baker.make(RewardPointRedemptionEvent, redeem_end_date=date.today() + timedelta(days=1))
+
+    def test_positive_points(self):
+        baker.make(RewardPointGranting, user_profile=self.student, value=5)
+        baker.make(RewardPointRedemption, user_profile=self.student, event=self.event, value=3)
+
+        response = self.app.get(self.url, user=self.test_users[0], status=200).content
+        self.assertTrue("student@institution.example.com;2" in str(response))
```
[the remark this answers is not in the document]
Comment (PFischbeck) [reply evap/rewards/tests/test_views.py]: Done, thanks!

File: evap/rewards/views.py (hunk 1 above)
Comment (PFischbeck) [reply evap/rewards/views.py]: Done!

**Review by janno42 — APPROVED** [review]

**Review by richardebeling — APPROVED** [review]
thanks! :)

**Pull request merged**